src/firefly/js/visualize/WorkspaceCntlr.js

```diff
@@ -258,6 +258,26 @@ export function getWorkspaceStatus() {
     return isEmpty(getWorkspaceList()) && get(flux.getState(), [WORKSPACE_PATH, 'status']);
 }
 
+export function getWorkspaceErrorMsg(){
+    const statusMsg = getWorkspaceStatus();
+    //The status code and corresponding messages are in ResponseMessage.java
+    var errorMsg;
+    switch (statusMsg){
+        case 'Unauthorized':
```

Comment:
Isn't more generic to use a status code than the status message that can change anytime?

Reply:
I planned to change the server code to pass the status code instead of the status message.  But I was worried the impact to others.  Since only the login message is matter, so I gave up my first plan.  But if you want it, I can do so.